gc_tests/run.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2019 King's College London created by the Software Development Team
+// <http://soft-dev.org/>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0>, or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, or the UPL-1.0 license <http://opensource.org/licenses/UPL>
+// at your option. This file may not be copied, modified, or distributed except according to those
+// terms.
+
+/// This is a simple test runner for testing the output of Rust programs linked
+/// with GC at run-time. Each ".rs" file in this directory will be compiled with
+/// rustcgc, and linked with GC runtime. Each file must start with a rustdoc
+/// comment in the following format:
+///
+/// ```text
+///   <type>:
+///     <body>
+/// ```
+///
+/// where:
+///
+///   * `<type>` is one of "Compile-time error", "Run-time error", or "Run-time
+///   output".
+///   * `<body>` is fuzzy text which will be matched against the Rust program's
+///   output literally except the following three exceptions:
+///       * All leading/trailing whitespace in both `<body>` and the Rust
+///         programs's output is ignored.
+///       * A line consisting solely of `...` is a wildcard which skips as many
+///         lines as necessary in order to match the subsequent line. It is an
+///         error to have two consecutive lines consisting of `...`.
+///       * A line starting with `...` but ending in non-whitespace matches the
+///         end portion of that string.
+use std::{
+    env,
+    fs::{read_dir, read_to_string},
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::{self, Command}
+};
+
+use lazy_static::lazy_static;
+use regex::{Regex, RegexBuilder};
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+const RUST_EXT: &'static str = "rs";
+const TESTS_DIR: &'static str = "gc_tests";
+const WILDCARD: &'static str = "...";
+const RUN_FILE: &'static str = "run.rs";
+const TEST_OBJ_PATH: &'static str = "TEST_OBJ_PATH";
+
+enum ExpType {
+    CompileTimeError,
+    RunTimeError,
+    RunTimeOutput
+}
+
+lazy_static! {
+    static ref EXPECTED: Regex = RegexBuilder::new(r#"^(/{3}(.*?)\n)*^"#)
```

Comment:
The regex differs from that in `yksom`, as we are looking for rustdoc comments in the `.rs` source file.